tests/pylint/test_enforce_type_hints.py

```diff
@@ -737,3 +743,55 @@ def capability_attributes(
 
     with assert_no_messages(linter):
         type_hint_checker.visit_classdef(class_node)
+
+
+def test_invalid_device_class(
+    linter: UnittestLinter, type_hint_checker: BaseChecker
+) -> None:
+    """Ensure invalid hints are rejected for entity device_class."""
+    # Set bypass option
+    type_hint_checker.config.ignore_missing_annotations = False
+
+    class_node, prop_node = astroid.extract_node(
+        """
+    class Entity():
+        pass
+
+    class CoverEntity(Entity):
+        pass
+
+    class MyCover( #@
+        CoverEntity
+    ):
+        @property
+        def device_class( #@
+            self
+        ):
+            pass
+    """,
+        "homeassistant.components.pylint_test.cover",
+    )
+    type_hint_checker.visit_module(class_node.parent)
+
+    with assert_adds_messages(
```

Comment:
I managed to get StrEnum to work, but I am not quite sure yet how we can avoid having two messages here.
I've added the test to show it, and maybe if we find a way to avoid it later we can clean the test accordingly...

Reply:
I managed to get rid of the duplicate on device_class, and dropped the StrEnum workaround.